go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -59,17 +61,70 @@ func transformToLogicalPlan(ctx *plancontext.PlanningContext, op ops.Operator, i
 		return transformCorrelatedSubQueryPlan(ctx, op)
 	case *operators.Derived:
 		return transformDerivedPlan(ctx, op)
-	case *operators.SimpleProjection:
-		return transformSimpleProjection(ctx, op)
 	case *operators.Filter:
 		return transformFilter(ctx, op)
 	case *operators.Horizon:
 		return transformHorizon(ctx, op, isRoot)
+	case *operators.Projection:
+		return transformProjection(ctx, op)
 	}
 
 	return nil, vterrors.VT13001(fmt.Sprintf("unknown type encountered: %T (transformToLogicalPlan)", op))
 }
 
+func transformProjection(ctx *plancontext.PlanningContext, op *operators.Projection) (logicalPlan, error) {
+	src, err := transformToLogicalPlan(ctx, op.Source, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if cols := op.AllOffsets(); cols != nil {
+		// if all this op is doing is passing through columns from the input, we
+		// can use the faster SimpleProjection
+		return simplifyPlan(op, cols, src)
+	}
+
+	expressions := slices2.Map(op.Columns, func(from operators.ProjExpr) sqlparser.Expr {
+		return from.GetExpr()
+	})
+
+	return &projection{
+		source:      src,
+		columnNames: op.ColumnNames,
+		columns:     expressions,
+	}, nil
+}
+
+func simplifyPlan(op *operators.Projection, cols []int, src logicalPlan) (logicalPlan, error) {
+	// all columns are just passing through something from below
```

Comment:
I might misunderstand this comment, but I think it is misplaced? The code below simply return an error.